fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/prunesubfield/SubfieldAccessPathNormalizer.java

```diff
@@ -154,11 +165,77 @@ public Optional<AccessPath> visitCall(CallOperator call, List<Optional<AccessPat
                     || FunctionSet.ARRAY_LENGTH.equals(call.getFnName())) {
                 return childrenAccessPaths.get(0)
                         .map(p -> p.appendPath(ColumnAccessPath.PATH_PLACEHOLDER, TAccessPathType.OFFSET));
+            } else if (PruneSubfieldRule.SUPPORT_JSON_FUNCTIONS.contains(call.getFnName())
+                    && call.getArguments().size() > 1 && call.getArguments().get(1).isConstantRef()) {
+
+                String path = ((ConstantOperator) call.getArguments().get(1)).getVarchar();
+                // we flatten whole json path, and control the query hierarchy dynamically through BE-self
+                return childrenAccessPaths.get(0).map(p -> p.appendFieldNames(formatJsonPath(path)));
             }
 
             return Optional.empty();
         }
 
+        // format json path, same as BE's JsonPathPiece, just supported simple path for prune subfield
+        // split char: .
+        // escape char: \
+        // quota char: "
+        //
+        // eg.
+        //  $.a.b -> [a, b]
+        //  $.a[0].b -> [a] -- don't support array index
+        //  $."a.b".c -> ["a.b", c]
+        //  $.a#b.c -> [a#b, c]
+        //  $.a.b.c.d.e.f -> [a, b] -- don't support overflown JSON_FLATTEN_DEPTH
+        //  a.b.c -> [a, b, c]
+        // when meet some unsupported path, return null
+        public static List<String> formatJsonPath(String path) {
```

Comment:
How process scenes like specialal character in a key. For example a json value `{'$abc$' : 1}`

Reply:
normally, it should be written as `$.$abc$` in json function, but I think we don't support the pattern in BE.

here will return empty list because doesn't match the regular expression for JSON PATH, and don't prune JSON subfield